### PR TITLE
[MI-3420] Fix issue of attachments sent from MM to Teams not opening for receiver due to permissions issue

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Fix the issue of attachments sent from MM to Teams not opening for a receiver due to permissions issue

#### Ticket Link
#248

### CODE REVIEW

File: server/message_hooks_test.go [hunk 1]
```diff
@@ -1184,20 +1215,21 @@ func TestSendChat(t *testing.T) {
...
 					MSTeamsID:      "mockMessageID",
 				}).Return(nil).Times(1)
 			},
 			SetupClient: func(client *clientmocks.Client, uclient *clientmocks.Client) {
-				uclient.On("CreateOrGetChatForUsers", mock.Anything).Return("mockChatID", nil).Once()
-				uclient.On("UploadFile", "", "", "mockFile.Name"+"_"+testutils.GetID()+".txt", 1, "mockMimeType", bytes.NewReader([]byte("mockData"))).Return(nil, errors.New("unable to upload the attachments")).Times(1)
-				uclient.On("GetChatMessage", "mockChatID", "mockParentMessageID").Return(&msteams.Message{
+				uclient.On("CreateOrGetChatForUsers", mock.Anything).Return(testutils.GetChatID(), nil).Once()
+				uclient.On("GetChat", testutils.GetChatID()).Return(mockChat, nil).Times(1)
+				uclient.On("UploadFile", "", "", "mockFile.Name"+"_"+testutils.GetID()+".txt", 1, "mockMimeType", bytes.NewReader([]byte("mockData")), mockChat).Return(nil, errors.New("unable to upload the attachments")).Times(1)
```
Comment (raghavaggarwal2308) [inline server/message_hooks_test.go:1225]: why some fields are empty?

File: server/msteams/client.go [hunk 2]
```diff
@@ -541,7 +541,50 @@ func (tc *ClientImpl) UploadFile(teamID, channelID, filename string, filesize in
 			return nil, NormalizeGraphAPIError(err)
 		}
 
-		itemID = *rootDirectory.GetId() + ":/" + filename + ":"
+		chatFolderID := ""
```
Comment (raghavaggarwal2308) [inline server/msteams/client.go:544]: declare it below where its needed

File: server/msteams/client.go [hunk 3]
```diff
@@ -541,7 +541,50 @@ func (tc *ClientImpl) UploadFile(teamID, channelID, filename string, filesize in
...
+			folderRequestBody := models.NewDriveItem()
+			folderRequestBody.SetName(&folderName)
+			folder := models.NewFolder()
+			folderRequestBody.SetFolder(folder)
+			additionalData := map[string]interface{}{
+				"microsoftGraphConflictBehavior": "fail",
+			}
+
+			folderRequestBody.SetAdditionalData(additionalData)
+			chatFolder, cErr := tc.client.DrivesById(driveID).ItemsById(*rootDirectory.GetId()).Children().Post(tc.ctx, folderRequestBody, nil)
```
Comment (raghavaggarwal2308) [inline server/msteams/client.go:579]:
```suggestion
			chatFolder, err := tc.client.DrivesById(driveID).ItemsById(*rootDirectory.GetId()).Children().Post(tc.ctx, folderRequestBody, nil)
```

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/message_hooks_test.go (hunk 1 above)
Comment (ayusht2810) [reply server/message_hooks_test.go:1225]: those are teamID and channelID and test cases are for chat

File: server/msteams/client.go [hunk 4]
```diff
@@ -560,6 +603,27 @@ func (tc *ClientImpl) UploadFile(teamID, channelID, filename string, filesize in
...
 	}
 	defer res.Body.Close()
+
+	if chat != nil {
+		requestBody := drives.NewItemItemsItemInvitePostRequestBody()
+		recipients := []models.DriveRecipientable{}
+		for _, chatUser := range chat.Members {
+			driveRecipient := models.NewDriveRecipient()
+			email := chatUser.Email
+			driveRecipient.SetEmail(&email)
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/msteams/client.go:597]: code logic is not working correctly with this

File: server/msteams/client.go [hunk 5]
```diff
@@ -541,50 +541,34 @@ func (tc *ClientImpl) UploadFile(teamID, channelID, filename string, filesize in
...
-
-			folderRequestBody.SetAdditionalData(additionalData)
-			chatFolder, cErr := tc.client.DrivesById(driveID).ItemsById(*rootDirectory.GetId()).Children().Post(tc.ctx, folderRequestBody, nil)
-			if cErr != nil {
+				folderRequestBody.SetAdditionalData(additionalData)
+				chatFolder, cErr = tc.client.DrivesById(driveID).ItemsById(*rootDirectory.GetId()).Children().Post(tc.ctx, folderRequestBody, nil)
+				if cErr != nil {
+					return nil, NormalizeGraphAPIError(cErr)
+				}
+			} else {
```
Comment (manojmalik20) [inline server/msteams/client.go:566]: Reverse this if-else.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**